action.yaml

```diff
@@ -54,6 +54,46 @@ runs:
 
         fi
 
+    - name: Test Report
+      uses: dorny/test-reporter@v1
+      id : test-result
+      if: ${{ env.cmd_type == 'UNIT_TEST' &&  env.test_report_path !='' && env.reporter !='' }}                   
+      with:
+        name: Unit Test                          
+        path: ${{ env.test_report_path}}              # '**/target/**/*.xml'   
+        reporter: ${{ env.reporter }}                 # Format of test results. 
+        only-summary: 'true' 
+        fail-on-error: ${{ env.fail-on-error || false }}     # Set action as failed if test report contains any failed test
+
+    - name: Copy test report to target folder
+      if: ${{ env.copy-to-target-path !='' }}
+      run: |
+        echo "Copy report files to target folder......"
+        cp ${{ env.test_report_path}} ${{env.copy-to-target-path}}
+      shell: bash
+
+    - name: Fetch Unit Test Report Url 
+      if: ${{ always() }}
```

Comment:
Can't be just `always()`. Imagine that you running the build step only or just the sonar step. What would happen?  It would fail most likely

Reply:
When we set the parameter as `fail-on-error==true` in the` Test Report` step then test reports are getting generated but the  `Fetch Unit Test Report Url `  step will not be executed because the `Test Report` step is failing so avoid that behavior we added `always()`. 

Yes this step` Fetch Unit Test Report Url ` will run every time we call composite action but we have added this code snippet
 ```
url=""
if echo "$html_url" | grep -q "/${{ github.repository }}/runs/"; then
    url=$html_url
fi 
```
Inside of step` Fetch Unit Test Report Url ` that checks whether the generated url is a valid test-report or not. If not it will set the url="" in `GITHUB_OUTPUT`.

Reply:
at least `env.cmd_type == 'UNIT_TEST'` needs to be here also. IF someone just add the copy-to-target-path env variable while building - for instance - this step gets triggered also. We need to tie this to unit tests step only.

Reply:
Updated condition for `'Fetch Unit Test Report Url' ` step with `env.cmd_type == 'UNIT_TEST'`.

Reply:
I'm not seeing the change. Still seeing
```
 - name: Fetch Unit Test Report Url 
   if: ${{ always() }}
```

Reply:
Modified code with a shell script to get specific html URL for the unit test job also added condition for copy to target step as check if destination folder exists or not.